src/nvim/diff.c

```diff
@@ -1547,25 +1546,21 @@ static void diff_read(int idx_orig, int idx_new, diffio_T *dio)
   if (dout->dout_fname == NULL) {
     diffstyle = DIFF_UNIFIED;
   } else {
-    fd = os_fopen((char *)dout->dout_fname, "r");
+    fd = os_fopen(dout->dout_fname, "r");
     if (fd == NULL) {
       emsg(_("E98: Cannot read diff output"));
       return;
     }
   }
 
-  diffhunk_T *hunk = NULL;
-
-  if (!dio->dio_internal) {
-    hunk = xmalloc(sizeof(*hunk));
-  }
+  diffhunk_T hunk;
```

Comment:
The scope of `hunk` can now be reduced.

Reply:
Done as part of some more refactoring of hunk extraction.